### ISSUE TITLE
[non-session-manager] Some sessions aren't listed

### ISSUE THREAD

**Issue by nareto** [original-male/non#192]
Just updated my non-session-manager installation (commit 16885e69fe865495dc32d869d1454ab148b0dca6) and I'm getting a strange behaviour. Some sessions present in --session-root are not listed at all. It seems pretty random: if I create a new session (without adding any clients to it) close the GUI and relaunch it, I can't see that session anymore. 

I should mention this is happening in a git-annex tree (the session root path is a subdirectory of the git-annex root directory), and from a brief test it doesn't seem to happen in other "normal" directories. However I can't see how this could be a problem since: 
1. it's happening also on new sessions, which thus are not symlinks (git-annex gets aware of new files only when you do "git add" I believe)
2. it happens even if I do "git annex unlock" on the directories, which effectively copies the original files in place of the symlinks.

Is there any way I can get a more verbose output?

**Comment by nareto** [comment]
I tried checking out commits from last year (when I'm sure I didn't have this problem) and I still suffer from this. So it's probably not directly dependent on NSM.

Any ideas on how I might debug this? Starting to get desperate here, as I wouldn't know where to start or where to ask for help even.

**Comment by nico202** [comment]
Have you tried coping one of these in a new folder and run (say "~/newpath")
```non-session-manager --session-root ~/newpath```
?
also, can you share one of these? I'll check if mine opens it. Maybe they are missing file session.nsm

**Comment by original-male** [comment]
I haven't had a chance to look into this yet. But I believe the next step is just to add some debugging output in the session discovery routines. If you have a complex symlink arrangement, or perhaps some unusual file/directory permissions, I could see that being a problem. It could also be some unusual characters in file/directory names confusing NTK and causing the sessions to be discovered, but fail to appear in the list. Extra debugging output would help to determine if that's the case or not.

**Comment by original-male** [comment]
Also, yes, it's not considered an NSM session unless there's an session.nsm file in the directory.

**Comment by nareto** [comment]
yes I tried copying to another path. I also created another user, copied the session there and starting NSM as this new user with the correct path, I can't see the sessions. [Here](https://www.dropbox.com/s/fxi657uuvjah3ey/test_session.tar?dl=0) and [here](https://www.dropbox.com/s/19erdr82knl3v8o/test_session-ardour4.tar?dl=0) are two test sessions. The first one is empty, to the second one I added only ardour4. When I create them in the GUI I can see them in the list on the left, open, save and close them normally, then I quit and when I restart again they are not there. The folders seem to be created correclty on disk

**Comment by original-male** [comment]
@nareto Please post the output of ls -lh in the session directory of the test_session-ardour.

**Comment by nareto** [comment]
$ ls -lh
    total 8,0K
    drwxr-xr-x 9 renato renato 4,0K Apr 22 00:16 Ardour.nXBKW
    -rw-r--r-- 1 renato renato   52 Apr 22 00:16 session.nsm


I also realized some other stuff:
1. I hadn't recompiled ntk in a long time (and I'm on Arch so maybe some library it depends on got updated), so I just checked out the latest git commit. I also removed ~/.config/ntk. Nothing changes
2. This is now happening also if I create a session in any path, so I don't think it has anything to do with git-annex
3. There is some type of error line in the output of the command I noticed only now:
    $ non-session-manager -- --session-root .
    [non-session-manager] Starting daemon...
    X_ChangeProperty: BadValue (integer parameter out of range for operation) 0x0
    [nsmd] Session root is: .
    NSM_URL=osc.udp://localhost.localdomain:15006/
    [nsmd] Listing sessions
    [non-session-manager] Telling server to quit

**Comment by nico202** [comment]
both works for me... don't know how to help, sorry
(for the hd fail, try dmesg to see if you have I/O error)

**Comment by original-male** [comment]
@nareto The X_ChangeProperty error is not related. I suppose your drive could be failing. You should test it with smartctl -t short and/or -t long. However, if that were the case I wouldn't expect ls to show the files either. I'm not familiar with whatever filesystem trickery git-annex may involve...

**Comment by nareto** [comment]
I do actually have some weird stuff going on my drive. So I just ordered a new one, and I hope to report back in a week or so when I've done the migration.

**Comment by nareto** [comment]
changed hard drive and still had same problem. But now I have it working - I discovered there was this broken link corresponding to an escape sequence in an "externals" directory of ardour in one of my sessions

    renato@triad externals $ ll
    total 0
    lrwxrwxrwx 1 renato renato 1 May 29 11:06 ''$'\001' -> ''$'\001'
    renato@triad externals $ rm ^A 

removing it (and other similar to this) made the session listing work again. BTW I can reproduce this - adding such links messes up the session listing
